File: languages/R/src/model/UserInfo.R
# Criando uma classe "UserInfo"
UserInfo = 
  R6Class("UserInfo",
          # Definindo valores e funções públicas
          public = list(
            
            # Construtor da classe
            # user -> nome do usuário
            # password -> senha do usuário
            # credit -> credito do usuário
            initialize = function(user, password, credit) {
              private$user = user
              private$password = password
              private$credit = credit
            },
            
            # Método getter de "user"
            getUser = function() { return(private$user) },
            
            # Método getter de "credit"
            getCredit = function() { return(private$credit) },
            
            # Método getter customizado de "password"
            getPassword = function() { 
              crypt = private$cryptPassword( private$password ) 
              return(crypt) 
            },
            
            # Método setter de "user"
            setUser = function(user) { self$user = user },

            # Método getter de "credit"
            setCredit = function(credit) { self$credit = credit },
            
            # Método getter de "password"
            setPassword = function(password) { self$password = password }
            
          ),
          # Definindo valores e funções privadas
          private = list(
            user = NULL,
            password = NULL,
            credit = 0,
            
            # Método privado para encriptar senhas
            # password -> senha a ser encriptada
            # RETURN -> nova senha
            cryptPassword = function(password) {
              crypt = strsplit(password, "")[[1]]
              crypt = rev(crypt)
              return(sprintf("HASH%s000", paste(crypt, collapse = "") ))
            }
          )
 )

File: languages/java/src/box/UserInfo.java
package box;

/**
 * Construção da classe "UserInfo"
 */
public class UserInfo {

    // String privada "user"
    private String user;

    // String privada "password"
    private String password;

    /**
     * Construtor público da classe
     * @param user nome do usuario
     * @param password senha do usuario
     */
    public UserInfo(String user, String password) {
        this.user = user;
        this.password = password;
    }

    // Getter da variavel "user"
    public String getUser() {
        return user;
    }

    // Setter da variavel "user"
    public void setUser(String user) {
        this.user = user;
    }

    // Getter customizado da variavel "password"
    public String getPassword() {
        return this.cryptPassword(password);
    }

    // Setter da variavel "password"
    public void setPassword(String password) {
        this.password = password;
    }

    /**
     * Método para encriptar senhas
     * @param password senha a ser encriptada
     * @return senha encriptada
     */
    private String cryptPassword(String password) {
        char[] array = password.toCharArray();
        char[] reverse = new char[array.length];

        for(int i = 0; i > array.length; i++){
            int index = (array.length - 1) - i;
            reverse[index] = array[i];
        }

        String cryptPass = reverse.toString();
        return(cryptPass);
    }
}
